modules/game/src/main/Metadata.scala

```diff
@@ -31,12 +31,16 @@ private[game] object Metadata:
   val empty =
     Metadata(None, None, None, None, None, analysed = false, GameDrawOffers.empty, rules = Set.empty)
 
-case class GameDrawOffers(white: Set[Ply], black: Set[Ply]):
+case class GameDrawOffers(white: Set[Ply], black: Set[Ply], lastDrawColor: Option[Color]):
```

Comment:
isn't that new `lastDrawColor` superfluous? it seems it can be inferred from the existing values - highest ply of the 2 sets

Reply:
Hi,

If you want to limit at once _per ply_ per player a draw request, then yes I could change it to do that.

I thought about using the Sets too at first, but then, just imagine, 
On a ply "x" :
1- W offers draw, "x" is added to the white set. B decline the offer. 
2- B offers draw, "x" is added to the black set. W decline the offer.
3- Who can now offer ? Because _looking at the set, you have no way of knowing [server-side] who offered last_, since (the set maximums) are equals. How can we know which was last ? Then let's say we allow to offer when they are equals (max-ply-offer-W == max-ply-offer-B). Then we still don't have a way of knowing who did the last one server-side (and a player could be double-offering in a row). We could start _modifying_ the sets to represent that it is as if a draw was not done for a player during a ply, but this would cause other problems down the line for sure (plus as the draw offers get normalized/annotated [another issue I created, they aren't always networked/visible when reloading], and they would then be missing.

Imagine the scenario where the position is very complex. W sees it is drawish and offers a draw. B declines immediately. Then realized it is drawish and offers a draw. B then cancel as he thought he was wrong (OR W think he was wrong and decline the offer as he thought he was wrong initially). Then both people realized it's a draw. The game is now forced to progressed with two people wanting to draw. I am not sure this is wanted as explained via refering to the wiki in the issue.

Reply:
> If you want to limit at once _per ply_ per player a draw request, then yes I could change it to do that.

If we go this road, We should consider FIDE rule, that only allow offer the draw in your turn. But it also has some other special case I think.

But tf we don't want to do that (keep the current behavior), in case of both offer draw at a same Ply. We can make them wait til next play in order to play. How does that sound?

Reply:
I'm good for whatever; I'd just want the official ruling ^^  (and ideally not wait 20 plies for a 2nd offer).

Reply:
> in case of both offer draw at a same Ply. We can make them wait til next ply in order to offer another draw.

I prefer the this approach. It makes sense. keep the current behavior, and doesn't required big changes. So, I guess you can just go for it.